src/hooks/useUserStatusMessage.ts

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2021 Šimon Brandner <simon.bra.ag@gmail.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { RoomMember } from "matrix-js-sdk";
+import { User } from "matrix-js-sdk/src/models/user";
+import { useCallback, useEffect, useState } from "react";
+
+import { GroupMember } from "../components/views/right_panel/UserInfo";
+import { MatrixClientPeg } from "../MatrixClientPeg";
+import { useEventEmitter } from "./useEventEmitter";
+
+type StatusMessageUser = User | RoomMember | GroupMember;
+
+const getUser = (user: StatusMessageUser): User => MatrixClientPeg.get().getUser(user?.userId);
+const getStatusMessage = (user: StatusMessageUser): string => getUser(user).unstable_statusMessage;
+
+// Hook to simplify handling Matrix User status
+export const useUserStatusMessage = (user?: StatusMessageUser): string => {
+    const [value, setValue] = useState<string>(getStatusMessage(user));
+
+    const update = useCallback(() => {
+        if (!user) return;
+        setValue(getStatusMessage(user));
+    }, [user]);
+
+    useEventEmitter(getUser(user), "User.unstable_statusMessage", update);
+    useEffect(() => {
+        update();
+        return () => {
+            setValue(undefined);
+        };
+    }, [update]);
+    return value;
```

Comment:
```suggestion
    return useEventEmitterState(getUser(User), "User.unstable_statusMessage", () => {
        return getStatusMessage(user);
    });
```

Think this could be shortened

Reply:
Ah, cool :+1: